lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -3910,18 +3910,36 @@ void request_disassemble(const llvm::json::Object &request) {
     return;
   }
 
-  addr_ptr += GetSigned(arguments, "instructionOffset", 0);
-  lldb::SBAddress addr(addr_ptr, g_dap.target);
+  addr_ptr += GetSigned(arguments, "offset", 0);
+  lldb::SBAddress addr = g_dap.target.ResolveLoadAddress(addr_ptr);
   if (!addr.IsValid()) {
     response["success"] = false;
     response["message"] = "Memory reference not found in the current binary.";
     g_dap.SendJSON(llvm::json::Value(std::move(response)));
     return;
   }
 
-  const auto inst_count = GetUnsigned(arguments, "instructionCount", 0);
-  lldb::SBInstructionList insts =
-      g_dap.target.ReadInstructions(addr, inst_count);
+  int64_t inst_offset = GetSigned(arguments, "instructionOffset", 0);
+  const int64_t inst_count = GetSigned(arguments, "instructionCount", 0);
```

Comment:
Use the `llvm::Expected<int64_t> GetSigned(...)`? If this is required. Same thing for "instructionOffset" above.